### ISSUE TITLE
Move break functionality to an to an add-on

### ISSUE THREAD

**Issue by thc202** [zaproxy/zaproxy#8153]
The break functionality does not need be in core and being in an add-on allows to enhance/fix it without the need to wait for a full release.

**Comment by thc202** [comment]
Do you know what needs to be done?

**Comment by aryangupta701** [comment]
Yes I have a top view idea of this. We need to create a new add on in zap-extensions and use that in zaproxy using interfaces and remove the actual break implementation in core. As I research more on it, I will take your help if stuck somewhere.

**Comment by thc202** [comment]
Not quite for the second part, in the end there will be no break functionality in core (i.e. no interfaces needed). Move the functionality to an add-on and change existing add-ons to use it. We should also discuss if we want third-party add-ons to keep working while the transition happens.

**Comment by thc202** [comment]
Worth noting that there are core PRs changing the break functionality.

**Comment by aryangupta701** [comment]
got it. thank you
